mobile/library/common/engine.h

```diff
@@ -118,6 +118,14 @@ class Engine : public Logger::Loggable<Logger::Id::main> {
 
 private:
   envoy_status_t main(std::unique_ptr<Envoy::OptionsImplBase>&& options);
+  /**
+   * Dump Envoy stats into the provided envoy_data
+   * @params envoy_data which will be filed with referenced stats dumped in Envoy's standard text
+   * format.
+   * @return failure status if the engine is no longer running.
+   * This can be called from any thread, but will block on engine-thread processing.
+   */
+  envoy_status_t dumpStats(envoy_data* out);
```

Comment:
question: does this `dumpStats` method need to use `envoy_data`?  Can it just write to a std::string directly?

Reply:
awesome thanks for doing that!